docs/adrs/0549-composite-run-steps.md

```diff
@@ -251,8 +251,27 @@ If any of the steps fail in the composite action and the `continue-on-error` is
 For the composite action steps, it follows the same logic as above. In this example, `"Hello World 2"` will be outputted because the previous step has `continue-on-error` set to `true` although that previous step errored. 
 
 ### Defaults
+We will not support "defaults" in a composite action. 
 
-The composite action author will be required to set the `shell` and `workingDir` of the composite action. Moreover, the composite action author will be able to explicitly set the shell for each composite run step. The workflow author will not have the ability to change these attributes. 
+### Shell and Working-directory
+For each run step in a composite action, the action author can set the `shell` and `working-directory` attributes for that step. These attributes are optional for each run step - by default, the `shell` is set to whatever default value is associated with the runner os (ex: bash =\> Mac). Moreover, the composite action author can map in values from the `inputs` for it's `shell` and `working-directory` attributes at the step level for an action. 
```

Comment:
> These attributes are optional for each run step - by default, the `shell` is set to whatever default value is associated with the runner os

This [doesn't appear to be true](https://github.com/Andrew-Test-Org/Public-Test-Repo/runs/7140706556?check_suite_focus=true). Removing the `shell` attribute from any `run` step of a composite action results in:

> The template is not valid ... Required property is missing: shell